lib/ddtrace/sampler.rb

```diff
@@ -213,13 +213,18 @@ def sample!(span)
       span.sampled = pre_sample?(span) ? @pre_sampler.sample!(span) : true
 
       if span.sampled
-        # If priority sampling has already been applied upstream, use that, otherwise...
-        unless priority_assigned_upstream?(span)
-          # Roll the dice and determine whether how we set the priority.
-          priority = priority_sample!(span) ? Datadog::Ext::Priority::AUTO_KEEP : Datadog::Ext::Priority::AUTO_REJECT
+        # If priority sampling has already been applied upstream, use that value.
+        return span.sampled if priority_assigned?(span)
 
-          assign_priority!(span, priority)
-        end
+        # Check with post sampler how we set the priority.
+        sample = priority_sample!(span)
+
+        # Check if post sampler has already assigned a priority.
+        return span.sampled if priority_assigned?(span)
```

Comment:
Minor: Can the value of `span.sampled` be mutated as this method is being executed? If not, could we simply the early returns with `return true if priority_assigned?(...)` to clarify what we expect from it?

Reply:
It cannot, I simplified it was suggested. 👍